build.gradle.kts

```diff
@@ -5,323 +5,325 @@ import kotlin.collections.ArrayList
-    implementation("javax.help:javahelp:2.0.05")
-    implementation("com.fifesoft:rsyntaxtextarea:3.1.2")
-    implementation("net.sf.nimrod:nimrod-laf:1.2")
-    implementation("org.drjekyll:colorpicker:1.3")
-    implementation("org.drjekyll:fontchooser:2.4")
-    implementation("at.swimmesberger:swingx-core:1.6.8")
-    implementation("org.scijava:swing-checkbox-tree:1.0.2")
-    implementation("org.slf4j:slf4j-api:1.7.30")
-    implementation("org.slf4j:slf4j-simple:1.7.30")
-    implementation("com.formdev:flatlaf:1.2")
+  implementation("org.hamcrest:hamcrest:2.2")
+  implementation("javax.help:javahelp:2.0.05")
+  implementation("com.fifesoft:rsyntaxtextarea:3.1.2")
+  implementation("net.sf.nimrod:nimrod-laf:1.2")
+  implementation("org.drjekyll:colorpicker:1.3")
+  implementation("org.drjekyll:fontchooser:2.4")
+  implementation("at.swimmesberger:swingx-core:1.6.8")
+  implementation("org.scijava:swing-checkbox-tree:1.0.2")
+  implementation("org.slf4j:slf4j-api:1.7.30")
+  implementation("org.slf4j:slf4j-simple:1.7.30")
+  implementation("com.formdev:flatlaf:1.2")
 
-    // NOTE: Be aware of reported issues with Eclipse and Batik
-    // See: https://github.com/logisim-evolution/logisim-evolution/issues/709
-    // implementation("org.apache.xmlgraphics:batik-swing:1.14")
+  // NOTE: Be aware of reported issues with Eclipse and Batik
+  // See: https://github.com/logisim-evolution/logisim-evolution/issues/709
+  // implementation("org.apache.xmlgraphics:batik-swing:1.14")
 
-    testImplementation("org.junit.vintage:junit-vintage-engine:5.7.1")
+  testImplementation("org.junit.vintage:junit-vintage-engine:5.7.1")
 }
 
 java {
-    sourceCompatibility = JavaVersion.VERSION_14
-    targetCompatibility = JavaVersion.VERSION_14
+  sourceCompatibility = JavaVersion.VERSION_14
+  targetCompatibility = JavaVersion.VERSION_14
 }
 
 task<Jar>("sourcesJar") {
-    group = "build"
-    description = "Creates a source jar archive."
-    dependsOn.add("classes")
-    classifier = "src"
+  group = "build"
+  description = "Creates a source jar archive."
+  dependsOn.add("classes")
+  classifier = "src"
 
-    from(sourceSets.main.get().allSource)
+  from(sourceSets.main.get().allSource)
 }
 
 extra.apply {
-   val df = SimpleDateFormat("yyyy")
-   val year = df.format(Date())
-   val javaHome = System.getProperty("java.home") ?: throw GradleException("java.home is not set")
-   val cmd = javaHome + File.separator + "bin" + File.separator + "jpackage"
-   val jpackagecmd = if (cmd.contains(" ")) "\"" + cmd + "\"" else cmd
-   val parameters = ArrayList<String>(Arrays.asList(
-      jpackagecmd,
+  val df = SimpleDateFormat("yyyy")
+  val year = df.format(Date())
+  val javaHome = System.getProperty("java.home") ?: throw GradleException("java.home is not set")
+  val cmd = javaHome + File.separator + "bin" + File.separator + "jpackage"
+  val jPackageCmd = if (cmd.contains(" ")) "\"" + cmd + "\"" else cmd
```

Comment:
What about quoting unconditionally? It won't hurt

Reply:
I didn't write that construct. I don't know what platform it is intended to address.

I did just try it on my Mac with the quotes in the string. I get:
Execution failed for task ':createApp'.
> java.io.IOException: Cannot run program ""/Library/Java/JavaVirtualMachines/jdk-17.jdk/Contents/Home/bin/jpackage"": error=2, No such file or directory

Apparently one of those quotes is part of the error message. If i just set: jPackageCmd = "thisIsJunk", it prints it with one set of quotes.

If you can check that the quotes are not needed anywhere, I'd be willing to remove them altogether. I don't have all the platforms to check.

Reply:
I have verified that on my Mac, the original code fails anyway. If I put a jdk in a path containing a space, it doesn't work. It gives a message similar to the above.

It does work with a space in the path if I do not add quotes.

Reply:
> It does work with a space in the path if I do not add quotes.

That's looks odd, if you ask me. I will take a look too.

Reply:
I think the reason is that the command is being used by Java's ProcessBuilder. ProcessBuilder takes the command and arguments in an array; therefore, it isn't parsing such that those quotes would make sense. If I understand this right, the actual code would fail on every Unix-like system and maybe also on Windows.